packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,482 @@
+/// This widget is similar to [Flex] and also takes [textDirection] and
+/// [verticalDirection] parameters, which are used for deciding in what order
+/// the panes are laid out (e.g [TextDirection.ltr] would position [startPane]
+/// on the left and [endPane] on the right).
+///
+/// The [panePriority] parameter can be used to display only one pane on screens
+/// without any separating [DisplayFeature], by using [TwoPanePriority.start]
+/// or [TwoPanePriority.end]. When [TwoPanePriority.both] is used or when the
+/// screen has a separating [DisplayFeature], both panes are visible.
+///
+/// Similarly to [SafeArea] and [DisplayFeatureSubScreen], this widget assumes
+/// there is no distance between it and the first [MediaQuery] ancestor. If this
+/// is not true, the bounds of display features will not align with the
+/// separation between the two panes. The [inset] parameter can be used to align
+/// TwoPane with the real position of display features. The [inset] parameter is
+/// the padding between TwoPane and the edges of the screen.
+///
+/// Pane widgets are wrapped in modified [MediaQuery] parents, removing padding,
+/// insets and display features that no longer intersect with them.
+///
+/// If not provided, [textDirection] defaults to the ambient [Directionality].
+/// If none are provided, then the widget asserts during build in debug mode.
+/// The resolved [direction] of TwoPane is [Axis.horizontal] when there are
+/// display features separating the screen into two horizontal sub-screens, in
+/// which case the provided [direction] is ignored.
+///
+/// See also
+///
+///  * [DisplayFeature] and [MediaQueryData.displayFeatures], to further
+///    understand display features
+///  * [MediaQueryData.removeDisplayFeatures] which is used to remove padding,
+///    insets and display features for each pane.
+class TwoPane extends StatelessWidget {
+  /// Create a layout that shows two pane widgets side by side.
+  const TwoPane({
+    Key? key,
+    required this.startPane,
+    required this.endPane,
+    this.paneProportion = 0.5,
+    this.textDirection,
+    this.verticalDirection = VerticalDirection.down,
+    this.direction = Axis.horizontal,
+    this.panePriority = TwoPanePriority.both,
+    this.inset = EdgeInsets.zero,
+  }) : super(key: key);
+
+  /// The first pane.
+  ///
+  /// On a horizontal layout, where [direction] is [Axis.horizontal]:
+  ///
+  ///   * The first pane is on the left when [textDirection] is
+  ///     [TextDirection.ltr]
+  ///   * The first pane is on the right when [textDirection] is
+  ///     [TextDirection.rtl]
+  ///
+  /// On a vertical layout, where [direction] is [Axis.vertical]:
+  ///
+  ///   * The first pane is at the top when [verticalDirection] is
+  ///     [VerticalDirection.down]
+  ///   * The first pane is at the bottom when [verticalDirection] is
+  ///     [VerticalDirection.up]
+  ///
+  /// If [panePriority] is [TwoPanePriority.start], this is the only pane
+  /// visible.
```

Comment:
Why? What happens to the second pane then?

Reply:
If TwoPanePriority is start or end, then only one pane is part of the tree. In this case endPane is not part of the tree.